Display of non-owned value is broken

```rust
use stylist::style;

struct MyColors {
    default: String,
}

fn main() {
    let colors = MyColors {
        default: "white".to_string(),
    };
    let colors = &colors;

    format!("color: {color};", color = colors.default); // success
    style!(r#"color: ${color};"#, color = colors.default); // fail
}
```

```
error[E0507]: cannot move out of `colours.default` which is behind a shared reference
  --> src/main.rs:14:43
   |
14 |     style!(r#"color: ${color};"#, color = colors.default); // fail
   |                                           ^^^^^^^^^^^^^^ move occurs because `colors.default` has type `String`, which does not implement the `Copy` trait

```

--- a/packages/stylist-macros/src/output/cow_str.rs
+++ b/packages/stylist-macros/src/output/cow_str.rs
@@ -22,8 +22,10 @@ impl OutputCowString {
         let expr = expr.into_token_stream(&mut inner_context);
         inner_context.uses_static(); // .to_string().into()
         Self::Raw(
-            quote_spanned! {source.span()=>
-                (&{ #expr } as &dyn ::std::fmt::Display).to_string().into()
+            quote_spanned! {source.span() =>
+                ::std::convert::Into::<_>::into(
+                    ::std::string::ToString::to_string(&(#expr) as &dyn ::std::fmt::Display)
+                )
             },
             inner_context,
         )

--- a/packages/stylist/tests/inline_integrations/no_display_impl-fail.stderr
+++ b/packages/stylist/tests/inline_integrations/no_display_impl-fail.stderr
@@ -6,4 +6,4 @@ error[E0277]: `NoDisplay` doesn't implement `std::fmt::Display`
   |
   = help: the trait `std::fmt::Display` is not implemented for `NoDisplay`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
-  = note: required for the cast to the object type `dyn std::fmt::Display`
+  = note: required because of the requirements on the impl of `ToString` for `NoDisplay`

--- a/packages/stylist/tests/inline_integrations/no_display_impl-fail.stderr
+++ b/packages/stylist/tests/inline_integrations/no_display_impl-fail.stderr
@@ -1,9 +1,9 @@
 error[E0277]: `NoDisplay` doesn't implement `std::fmt::Display`
- --> $DIR/no_display_impl-fail.rs:7:23
+ --> tests/inline_integrations/no_display_impl-fail.rs:7:23
   |
 7 |         background: ${expr};
   |                       ^^^^ `NoDisplay` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `NoDisplay`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
-  = note: required because of the requirements on the impl of `ToString` for `NoDisplay`
+  = note: required for the cast to the object type `dyn std::fmt::Display`

--- a/packages/stylist/tests/literal_css_referenced_value.rs
+++ b/packages/stylist/tests/literal_css_referenced_value.rs
@@ -1,0 +1,31 @@
+#[test]
+fn test_sheet_value_by_ref() {
+    struct Theme {
+        color: String,
+    }
+
+    let theme = Theme {
+        color: "red".into(),
+    };
+
+    let theme = &theme;
+
+    use stylist::*;
+    let parsed = css!(
+        r#"
+            color: ${color};
+        "#,
+        color = &theme.color,
+    );
+
+    let style = Style::new(parsed).expect("Failed to parse style.");
+
+    let expected = format!(
+        r#".{cls} {{
+    color: red;
+}}
+"#,
+        cls = style.get_class_name()
+    );
+    assert_eq!(style.get_style_str(), expected);
+}

--- a/packages/stylist/tests/literal_css_referenced_value.rs
+++ b/packages/stylist/tests/literal_css_referenced_value.rs
@@ -15,7 +15,7 @@ fn test_sheet_value_by_ref() {
         r#"
             color: ${color};
         "#,
-        color = &theme.color,
+        color = theme.color,
     );
 
     let style = Style::new(parsed).expect("Failed to parse style.");

